### PR TITLE
feat: Refactored command for adding Function-Call key

### PR DESCRIPTION
**Pull request by FroVolod**

### CODE REVIEW

File: src/commands/transaction/construct_transaction/add_action_1/add_action/add_key/access_key_type/mod.rs [hunk 1]
```diff
@@ -100,13 +100,11 @@ impl FunctionCallType {
...
             Yes,
-            #[strum(
-                to_string = "No, I don't want to input a list of method names that can be used"
-            )]
+            #[strum(to_string = "No, I allow it to perform any methods from the contract")]
             No,
         }
         let select_choose_input = Select::new(
-            "Do You want to input a list of method names that can be used?",
+            "Do you want to limit the use of the \"function call key\" to only certain methods or allow it to perform any method according to the specified contract?",
```
Comment (frol) [inline src/commands/transaction/construct_transaction/add_action_1/add_action/add_key/access_key_type/mod.rs:107]:
```suggestion
            "Would you like the access key to be valid exclusively for calling specific functions on the contract?",
```

File: src/commands/transaction/construct_transaction/add_action_1/add_action/add_key/access_key_type/mod.rs [hunk 2]
```diff
@@ -100,13 +100,11 @@ impl FunctionCallType {
         enum ConfirmOptions {
             #[strum(to_string = "Yes, I want to input a list of method names that can be used")]
```
Comment (frol) [inline src/commands/transaction/construct_transaction/add_action_1/add_action/add_key/access_key_type/mod.rs:101]:
```suggestion
            #[strum(to_string = "Yes, I want to input a list of function names that can be called when transaction is signed by this access key")]
```

File: src/commands/transaction/construct_transaction/add_action_1/add_action/add_key/access_key_type/mod.rs [hunk 3]
```diff
@@ -100,13 +100,11 @@ impl FunctionCallType {
...
             Yes,
-            #[strum(
-                to_string = "No, I don't want to input a list of method names that can be used"
-            )]
+            #[strum(to_string = "No, I allow it to perform any methods from the contract")]
```
Comment (frol) [inline src/commands/transaction/construct_transaction/add_action_1/add_action/add_key/access_key_type/mod.rs:103]:
```suggestion
            #[strum(to_string = "No, I allow it to call any functions on the specified contract")]
```

File: src/commands/transaction/construct_transaction/add_action_1/add_action/add_key/access_key_type/mod.rs [hunk 4]
```diff
@@ -132,26 +130,10 @@ impl FunctionCallType {
...
-        if let ConfirmOptions::Yes = select_choose_input {
-            let allowance_near_balance: crate::types::near_token::NearToken =
-                    CustomType::new("Enter an allowance which is a balance limit to use by this access key to pay for function call gas and transaction fees (example: 10NEAR or 0.5near or 10000yoctonear):")
-                        .prompt()?;
-            Ok(Some(allowance_near_balance))
-        } else {
-            Ok(None)
-        }
+        let allowance_near_balance: crate::types::near_token::NearToken =
+            CustomType::new("Enter an allowance which is a balance limit to use by this access key to pay for function call gas and transaction fees (example: 10NEAR or 0.5near or 10000yoctonear):")
```
Comment (frol) [inline src/commands/transaction/construct_transaction/add_action_1/add_action/add_key/access_key_type/mod.rs:134]:
```suggestion
            CustomType::new("Enter the allowance, a budget this access key can use to pay for transaction fees (example: 10NEAR or 0.5near or 10000yoctonear):")
```

File: src/js_command_match/dev_deploy.rs [hunk 5]
```diff
@@ -33,7 +33,7 @@ impl DevDeployArgs {
         if let Some(init_args) = &self.init_args {
             let mut initial_function = "new".to_string();
             if let Some(init_function) = &self.init_function {
-                initial_function = init_function.clone();
+                initial_function.clone_from(init_function);
             }
```
Comment (frol) [inline src/js_command_match/dev_deploy.rs:37]: This whole block can be removed and `initial_function` below can be replaced with:

```rust
self.init_function.as_ref().unwrap_or("new").to_owned()
```

**Review by frol — CHANGES_REQUESTED** [review]

File: src/js_command_match/dev_deploy.rs (hunk 5 above)
Comment (FroVolod) [reply src/js_command_match/dev_deploy.rs]: done

**Review by frol — APPROVED** [review]

**Pull request merged**